processing/src/main/java/io/druid/segment/data/IntermediateLongSupplierSerializer.java

```diff
@@ -131,8 +131,10 @@ private void makeDelegate() throws IOException
 
     try (DataInputStream tempIn = new DataInputStream(new BufferedInputStream(ioPeon.makeInputStream(tempFile)))) {
       delegate.open();
-      while (tempIn.available() > 0) {
+      int available = tempIn.available();
```

Comment:
Hmm, the original code looks kind of sketchy. "available" is:

> Returns an estimate of the number of remaining bytes that can be read (or skipped over) from this input stream without blocking by the next invocation of a method for this input stream.

But this method looks like it actually wants to read the entire file. I think it's a bug in the original code, and your fix does improve speed but really we should fix the bug.

How about reading `numInserted` longs instead of checking `available()`?